presto-main/src/main/java/io/prestosql/operator/scalar/time/TimeOperators.java

```diff
@@ -0,0 +1,298 @@
+import io.prestosql.spi.function.LiteralParameters;
+import io.prestosql.spi.function.ScalarOperator;
+import io.prestosql.spi.function.SqlNullable;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.type.AbstractLongType;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.type.Constraint;
+import org.joda.time.chrono.ISOChronology;
+
+import static io.prestosql.spi.StandardErrorCode.INVALID_CAST_ARGUMENT;
+import static io.prestosql.spi.function.OperatorType.ADD;
+import static io.prestosql.spi.function.OperatorType.CAST;
+import static io.prestosql.spi.function.OperatorType.EQUAL;
+import static io.prestosql.spi.function.OperatorType.GREATER_THAN;
+import static io.prestosql.spi.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static io.prestosql.spi.function.OperatorType.HASH_CODE;
+import static io.prestosql.spi.function.OperatorType.INDETERMINATE;
+import static io.prestosql.spi.function.OperatorType.IS_DISTINCT_FROM;
+import static io.prestosql.spi.function.OperatorType.LESS_THAN;
+import static io.prestosql.spi.function.OperatorType.LESS_THAN_OR_EQUAL;
+import static io.prestosql.spi.function.OperatorType.NOT_EQUAL;
+import static io.prestosql.spi.function.OperatorType.SUBTRACT;
+import static io.prestosql.spi.function.OperatorType.XX_HASH_64;
+import static io.prestosql.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.prestosql.spi.type.TimeType.MAX_PRECISION;
+import static io.prestosql.spi.type.TimeType.TIME;
+import static io.prestosql.type.DateTimes.MINUTES_PER_HOUR;
+import static io.prestosql.type.DateTimes.PICOSECONDS_PER_DAY;
+import static io.prestosql.type.DateTimes.PICOSECONDS_PER_HOUR;
+import static io.prestosql.type.DateTimes.PICOSECONDS_PER_MILLISECOND;
+import static io.prestosql.type.DateTimes.PICOSECONDS_PER_MINUTE;
+import static io.prestosql.type.DateTimes.PICOSECONDS_PER_SECOND;
+import static io.prestosql.type.DateTimes.SECONDS_PER_MINUTE;
+import static io.prestosql.type.DateTimes.parseTime;
+import static io.prestosql.type.DateTimes.rescaleWithRounding;
+import static io.prestosql.type.DateTimes.round;
+import static io.prestosql.type.DateTimes.scaleFactor;
+import static io.prestosql.util.DateTimeZoneIndex.getChronology;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class TimeOperators
+{
+    private TimeOperators() {}
+
+    @ScalarOperator(SUBTRACT)
+    @LiteralParameters("p")
+    @SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND)
+    public static long subtract(@SqlType("time(p)") long left, @SqlType("time(p)") long right)
+    {
+        long interval = left - right;
+
+        interval = rescaleWithRounding(interval, MAX_PRECISION, 3);
+
+        return interval;
+    }
+
+    @ScalarOperator(EQUAL)
+    @LiteralParameters("p")
+    @SqlType(StandardTypes.BOOLEAN)
+    @SqlNullable
+    public static Boolean equal(@SqlType("time(p)") long left, @SqlType("time(p)") long right)
```

Comment:
I guess  this could be declared as returning `boolean`, not `Boolean`

(same below)

Reply:
No, the implementation of the equal operator needs to return Boolean (long story)